BootcampKotlin/app/src/main/java/com/github/ybecker/bootcampKotlin/MainActivity.kt

```diff
@@ -0,0 +1,53 @@
+package com.github.ybecker.bootcampKotlin
+
+import android.content.Intent
+import android.os.Bundle
+import androidx.activity.compose.setContent
+import androidx.appcompat.app.AppCompatActivity
+import androidx.compose.foundation.layout.Column
+import androidx.compose.material3.Button
+import androidx.compose.material3.ExperimentalMaterial3Api
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextField
+import androidx.compose.runtime.*
+import androidx.compose.ui.tooling.preview.Preview
+
+class MainActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            NameAndButton()
+        }
+    }
+
+    /**
+     * Creates the input field for the name and the button to transition to the greeting activity
+     */
+    @OptIn(ExperimentalMaterial3Api::class)
+    @Preview
+    @Composable
+    fun NameAndButton() {
+        // We can't use "this" inside the onClick lambda as we would not get the same context
+        val context = this
+
+        // Used to align vertically the window components
+        Column(content = {
```

Comment:
You don't need to specify "content ={...}" as it is the only argument that takes a (ColumnScope.() -> Unit). By removing it your code may look simpler.